crates/composefs/src/repository.rs

```diff
@@ -156,7 +156,20 @@ impl<ObjectID: FsVerityHashValue> Repository<ObjectID> {
         )?;
         drop(file);
 
-        if !self.insecure {
+        if self.insecure {
+            match measure_verity::<ObjectID>(&ro_fd) {
```

Comment:
The measure here doesn't make much sense.  We just created this, so the answer will always be either that there's no fs-verity on the file, or that fs-verity is unsupported.  But you could get that information by just trying to enable it and deal with the errors.  I think you could probably also merge the code paths with the `else` (original) case... The only different you really need is what to do if the enabling fails with the not-supported error: in secure mode, that's fatal.  In insecure mode, you ignore it.

Reply:
Also please merge this into the first commit which adds `Repository.insecure` in the first place.